javaparser-core-testing/src/test/java/com/github/javaparser/issues/Issue4786Test.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2013-2024 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.issues;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.github.javaparser.ParseResult;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.utils.SourceRoot;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Regression for issue 4786: when parsing via SourceRoot and a file fails to parse,
+ * diagnostics should indicate which file failed.
+ */
+public class Issue4786Test {
+
+    @Test
+    void failedParse_shouldIncludeFilePath_whenFileContainsBackslashUOnly() throws IOException {
+        // Arrange: create a temporary source root and a file whose content triggers a scanner failure
+        Path tempDir = Files.createTempDirectory("jpv-issue4786");
+        Path srcRoot = tempDir.resolve("src");
+        Files.createDirectories(srcRoot);
+
+        Path broken = srcRoot.resolve("OnlyUnicode.java");
+        // Write exactly two bytes: backslash and 'u'
+        Files.write(broken, new byte[] { (byte) '\\', (byte) 'u' });
+
+        SourceRoot sourceRoot = new SourceRoot(srcRoot);
+        sourceRoot.setParserConfiguration(new ParserConfiguration().setCharacterEncoding(StandardCharsets.UTF_8));
+
+        // Act
+        List<ParseResult<CompilationUnit>> results = sourceRoot.tryToParse("");
+
+        // Assert
+        assertEquals(1, results.size(), "Expected one parse result");
+        ParseResult<CompilationUnit> result = results.get(0);
+        assertFalse(result.isSuccessful(), "Parsing should fail for this content");
+        assertFalse(result.getProblems().isEmpty(), "Problems should be present on failure");
+
+        String diag = result.toString();
+        String expectedPath = broken.toAbsolutePath().toString();
+        assertTrue(diag.contains(expectedPath),
```

Comment:
The check must be based on a getSourcePath method that does not currently exist.